src/unix/lwt_unix.cppo.ml

```diff
@@ -136,6 +136,16 @@ let auto_yield timeout =
     end else
       Lwt.return_unit
 
+let auto_pause timeout =
```

Comment:
I would suggest not to add this function. It seems to me like users can just keep using `auto_yield`. `auto_pause` with timeouts is a little weird in meaning, because the whole point of `pause` is that it does not depend on `Lwt_unix`. However, `auto_pause` inherently depends on `Lwt_unix`, so why not just leave `auto_yield` only?

Reply:
The difference is whether eventually you would call `pause` or `yield` when the timeout happens. If secretly changing `yield` to `pause` inside `auto_yield` is okay, then I would argue we should simply silently replacing all `yield` with `pause` without deprecation. I believe our previous discussion has ruled out silent changes, and thus we need a different function is to give the users some warnings. Thus, we cannot leave `auto_yield` only.

The remaining objection seems to be the weirdness of its meaning. I disagree with using `pause` means that it has no place in `Lwt_unix`. In the `Unix` module of the standard library, for example, there's `Unix.in_channel_of_descr` to create `in_channel`. The point of `in_channel` is to be platform-independent, but that does not prevent specific platforms or backends to provide more functionality using the common mechanism. Here, `auto_pause` is extending the common `pause` using features on the specific platform.

Reply:
Also, after this PR, there would be no functions with the word `yield` in the lwt library except `Lwt_main.abandon_yielded_and_paused` (which would also be deprecated once `Lwt_main.yield` is completely gone), so it might actually be more consistent to call everything `pause`.